src/ray/gcs/pubsub/gcs_pub_sub.cc

```diff
@@ -513,6 +524,34 @@ bool GcsSubscriber::IsTaskLeaseUnsubscribed(const TaskID &id) {
 
 Status GcsSubscriber::SubscribeAllWorkerFailures(
     const ItemCallback<rpc::WorkerDeltaData> &subscribe, const StatusCallback &done) {
+  if (subscriber_ != nullptr) {
+    // GCS subscriber.
+    auto subscribe_item_callback = [subscribe](const rpc::PubMessage &msg) {
+      RAY_CHECK(msg.channel_type() == rpc::ChannelType::GCS_WORKER_DELTA_CHANNEL);
+      subscribe(msg.worker_delta_message());
+    };
+    auto subscription_failure_callback = [](const std::string &, const Status &status) {
+      RAY_LOG(WARNING) << "Subscription to WorkerDelta channel failed: "
+                       << status.ToString();
+    };
+    if (!subscriber_->SubscribeChannel(
+            std::make_unique<rpc::SubMessage>(),
+            rpc::ChannelType::GCS_WORKER_DELTA_CHANNEL, gcs_address_,
+            [done](Status status) {
+              if (done != nullptr) {
+                done(status);
+              }
+            },
+            std::move(subscribe_item_callback),
```

Comment:
here and below as well

Reply:
I think the variable names `subscribe_item_callback` and `subscription_failure_callback` are descriptive enough? Otherwise it could be overkill.

Reply:
Yeah that makes sense!